std/math.d

```diff
@@ -5160,52 +5194,76 @@ private:
             */
            assert(0, "Not yet supported");
         }
-        else version (ARM)
-        {
-            assert(false, "Not yet supported.");
-        }
         else version (RISCV_Any)
         {
-            mixin(`
-            uint result = void;
-            asm pure nothrow @nogc
-            {
-                "frflags %0" : "=r" (result);
-            }
+            uint result;
+            asm pure nothrow @nogc { "frflags %0" : "=r" (result); }
             return result;
-            `);
         }
         else
             assert(0, "Not yet supported");
     }
 
     static void resetIeeeFlags() @trusted
     {
-        version (InlineAsm_X86_Any)
+        version (X86_Any)
         {
-            asm nothrow @nogc
-            {
-                fnclex;
-            }
+            version (DigitalMars)
+                asm nothrow @nogc { fnclex; }
+            else
+                asm nothrow @nogc { "fnclex" : : : "fpsw"; }
 
             // Also clear exception flags in MXCSR, SSE's control register.
             if (haveSSE)
             {
                 uint mxcsr;
-                asm nothrow @nogc { stmxcsr mxcsr; }
-                mxcsr &= ~EXCEPTIONS_MASK;
-                asm nothrow @nogc { ldmxcsr mxcsr; }
+                version (DigitalMars)
+                {
+                    asm nothrow @nogc { stmxcsr mxcsr; }
+                    mxcsr &= ~EXCEPTIONS_MASK;
+                    asm nothrow @nogc { ldmxcsr mxcsr; }
+                }
+                else
+                {
+                    asm nothrow @nogc { "stmxcsr %0" : "=m" (mxcsr); }
+                    mxcsr &= ~EXCEPTIONS_MASK;
+                    asm nothrow @nogc { "ldmxcsr %0" : : "m" (mxcsr) : "flags"; }
```

Comment:
Unsure about the correct clobber for these x87 (`flags`?) and SSE flags (`fpsw`?) - see https://github.com/llvm-mirror/clang/blob/aa231e4be75ac4759c236b755c57876f76e3cf05/lib/Parse/ParseStmtAsm.cpp#L636-L640...

Reply:
glibc uses these clobber flags, so I trust them.

Reply:
e.g:
```
unsigned int xnew_exc;
/* Get the current control word.  */
__asm__ ("stmxcsr %0" : "=m" (*&xnew_exc));
xnew_exc &= ~(excepts << 7);
__asm__ ("ldmxcsr %0" : : "m" (*&xnew_exc));
```